pepys_import/resolvers/command_line_resolver.py

```diff
@@ -200,11 +213,40 @@ def is_valid_dynamic(option):  # pragma: no cover
                 privacy,
                 change_id,
             )
-        elif 3 <= int(choice) <= len(choices):
+        elif choice == str(3):
+            return self.add_unknown_platform(data_store, platform_name)
+        elif choice == str(4):
+            self.store_all_platforms_as_unknown = True
+            return self.add_unknown_platform(data_store, platform_name)
+        elif 5 <= int(choice) <= len(choices):
             # One of the pre-existing platforms was chosen
-            platform_index = int(choice) - 3
+            platform_index = int(choice) - 5
             return platforms[platform_index]
 
+    def add_unknown_platform(self, data_store, platform_name):
+        if platform_name is None:
+            platform_name = str(uuid.uuid4())
+        identifier = platform_name[:3]
```

Comment:
Aah, as before... I dont think we should produce the identifier by trimming the platform name.  If the IMO is being provided as `name` - then the full value is the best identifier we have, I think...

Reply:
Changed now.